### PR TITLE
Add ability to daemonize on Linux

### PR DESCRIPTION
**Pull request by zhindes**
### What does this Pull Request accomplish?

These changes allow the server process to properly daemonize on Linux including:
* Executing the complicated dance necessary to daemonize: double fork, closing std handles, etc.
* Creating a pid file
* Handling common signals and stopping the server when necessary
   * Provided a callback that will poke the server to shutdown
* Separated server startup into a couple distinct sections
   * Parse options and load configuration
   * Redirect to syslog/daemonize
   * Run server

### Why should this Pull Request be merged?

NI Sync is installing the grpc server as a boot-daemon on Linux RT. It is "branded" as `ni-sync-remote` to make it clear what functionality it is providing. Hence the identity options.

### What testing has been done?

Tested on a local system with an improved sync init script. Here is interleaved commands and syslog output (via `tail -f`)

```
admin@NI-PXIe-8880-03095FC1:~/grpc# /etc/init.d/ni-sync-remote start
2021-05-26T16:18:06.829+00:00 NI-PXIe-8880-03095FC1 ni-sync-remote[17888]: Server listening on port 31763
2021-05-26T16:18:06.829+00:00 NI-PXIe-8880-03095FC1 ni-sync-remote[17888]: Security is configured with insecure credentials.
admin@NI-PXIe-8880-03095FC1:~/grpc# /etc/init.d/ni-sync-remote status;echo $?
0
admin@NI-PXIe-8880-03095FC1:~/grpc# cat /var/run/ni-sync-remote.pid
17888
...
admin@NI-PXIe-8880-03095FC1:~/grpc# /etc/init.d/ni-sync-remote stop
admin@NI-PXIe-8880-03095FC1:~/grpc# 2021-05-26T16:28:41.805+00:00 NI-PXIe-8880-03095FC1 ni-sync-remote[17888]: handling signal: 15 (terminate)
admin@NI-PXIe-8880-03095FC1:~/grpc# /etc/init.d/ni-sync-remote status;echo $?
3
```

### CODE REVIEW

**Review by jonlugoatx — APPROVED** [review]

**Comment by zhindes** [comment]
@SparkingSpork - we have a few tasks blocked by this, and the changes to CMakeLists.txt are trivial, so I am going to complete without your approval. If you find any issues, I will fix them!

**Pull request merged**
